README.md

```diff
@@ -2,91 +2,131 @@
+docker run --user="$(id -u):$(id -g)" -it -v "$(PWD):/usr/src" sonarsource/sonar-scanner-cli -Dsonar.projectKey=<KEY>  -Dsonar.sources=.  -Dsonar.host.url=http://host.docker.internal:9000  -Dsonar.login=<STRING>
 ```
 
-### Write permissions
-
-The scanner writes to the analysed project's directory, in directory `${SONAR_PROJECT_BASE_DIR}/.scannerwork`.
-
-By default scanner writes with user id `1000` and group id `1000`. The `--user` option (see sample commands above) is used on Linux to have the scanner write with the same user and group as the caller to the `docker run` command.
-
-## On Mac
-
-The command is quite similar to the one for Linux except that you don't need to specify `--user` option.
+#### With SonarQube running in Docker
 
-To analyse the project located in `/path/to/project`, execute:
+Create a network and boot SonarQube:
 
 ```
-docker run -e SONAR_HOST_URL=http://foo.acme:9000 -it -v "/path/to/project:/usr/src" sonarsource/sonar-scanner-cli
+docker network create "scanner-sq-network"
+docker run --network="scanner-sq-network" --name="sq" -d sonarqube
 ```
 
-To analyse the project in the current directory, execute:
+And run the scanner:
 
 ```
-docker run -e SONAR_HOST_URL=http://foo.acme:9000 -it -v "$(pwd):/usr/src" sonarsource/sonar-scanner-cli
+docker run --user="$(id -u):$(id -g)" -it -v "$(PWD):/usr/src" --network="scanner-sq-network" sonarsource/sonar-scanner-cli -Dsonar.projectKey=<KEY>  -Dsonar.sources=.  -Dsonar.host.url=http://host.docker.internal:9000  -Dsonar.login=<STRING>
 ```
 
-## Supported environment variables
+## Write permissions
 
-### SonarQube host and authentication
+The scanner writes to the analysed project's directory,
+creating directories `${SONAR_PROJECT_BASE_DIR}/.scannerwork` and
+`${SONAR_PROJECT_BASE_DIR}/.sonar`.
 
-* `SONAR_HOST_URL`: URL to the SonarQube instance the scanner should connect to (default=`http://localhost:9000`)
-* `SONAR_TOKEN`: the prefered way to authenticate to SonarQube is to use a token. Use this environment variable to pass it to the scanner
-* `SONAR_LOGIN` and `SONAR_PASSWORD`: alternatively, you can provide the SonarQube username and password for the scanner to use using these two variables
+By default scanner writes with user id `1001` and group id `1001`. The `--user`
+option (see sample commands above) is used on Linux to have the scanner write
+with the same user and group as the caller to the `docker run` command.
 
 ### Project mounting point
 
 By default, the scanner analyses the project in directory `/usr/src`.
 
-If you can't mount the project into this directory (eg. in Gitlab CI), use `SONAR_PROJECT_BASE_DIR` to specify another location in the container.
+### Caching and scanner user home directory
 
-```
--e SONAR_PROJECT_BASE_DIR=/build
-```
+When running the scanner with this image, the `.sonar` directory is created in
+the project's directory. This implies caching is not happening accross analysis
+of multiple projects.
```

Comment:
I believe this isn't true anymore - `SONAR_USER_HOME` isn't set anywhere anymore (used to be in `entrypoint.sh`), so I believe the default behavior is using .sonar/ in USER_HOME directory?

Reply:
Wall of terminal output:
```
$ ls -1
Dockerfile
LICENSE
README.md
__pycache__
cacheview
images
renovate.json
requirements.txt
tests
tox.ini
$ docker run --user="$(id -u):$(id -g)" -it -v "$(PWD):/usr/src" sonarsource/sonar-scanner-cli -Dsonar.projectKey=ansible_cacheview  -Dsonar.sources=.  -Dsonar.host.url=http://host.docker.internal:9000  -Dsonar.login=52a56ed0d19c6283413ee1c07933c160c9f1da86
INFO: Scanner configuration file: /opt/sonar-scanner/conf/sonar-scanner.properties
INFO: Project root configuration file: NONE
INFO: SonarQube Scanner 4.2.0.1873
INFO: Java 11.0.5 Alpine (64-bit)
INFO: Linux 4.19.76-linuxkit amd64
**INFO: User cache: ?/.sonar/cache**
INFO: SonarQube server 7.9.2
INFO: Default locale: "en_US", source code encoding: "UTF-8" (analysis is platform dependent)
INFO: Load global settings
INFO: Load global settings (done) | time=149ms
INFO: Server id: BF41A1F2-AXAfx-3rzRBL7oqZsEle
**INFO: User cache: /usr/src/?/.sonar/cache**
INFO: Load/download plugins
INFO: Load plugins index
INFO: Load plugins index (done) | time=93ms
INFO: Load/download plugins (done) | time=18655ms
INFO: Process project properties
INFO: Execute project builders
INFO: Execute project builders (done) | time=7ms
INFO: Project key: ansible_cacheview
INFO: Base dir: /usr/src
**INFO: Working dir: /usr/src/.scannerwork**
INFO: Load project settings for component key: 'ansible_cacheview'
[... OUTPUT REMOVED ...]
$ ls -1
'?'
Dockerfile
LICENSE
README.md
__pycache__
cacheview
images
renovate.json
requirements.txt
tests
tox.ini
$ ls \?/.sonar/
_tmp  cache
$ ls \?/.sonar/cache/
0c3ec093156f0350edc21e2a1d7ef2ee  489f1b0ff745f9a744c23a97445caf98  59d98c94277e5faa8377ba521e440eba  84bac7bdcd3d78a2cd7037bfd70fcc3c	b8b93b70f10a74c6b3ab95ef8691c065
2d1a20d5ed66aa7f1273836a29fa08d4  4a3cc03335859f0839de090b0ed35956  6ead78e9824cd63c2952b34ee1f1231c  866bb1adbf016ea515620f1aaa15ec53	cb2c8bc61a566b6f9bd7f4a8d9518a25
2d9994a460180757f3fbad54f03e818e  511f59a24e74f40d7e810467f1e3a253  716a52eef1e2edefeafa4c83c9c33a22  90da8f56fbbbb3624b8a3f6f1075860a	e1977f8aaefcb2066437d32de0b347b0
37ae8929672c56a67919c60df53b4283  513697d7dc792f02b10b1f4bcb9b48ab  73d90ce62c95649d02af4292588b027d  91f25bbf08401b93d651d5f3653d2e0e
45e3bfcd65e7578ed4cd89604de3b06f  5246ba7166e8e9ab17d08a6514e9a086  74dfad9fb80c653ed5c959c3b2ed5a4a  _tmp
```

Reply:
`-Dsonar.projectBaseDir=/usr/src` is set as the Docker `CMD`, and the project directory is mounted into that directory in which the cache is written.

Reply:
oh, hah, I was wrong. `?/.sonar` 🤣!